mypyc/irbuild/constant_fold.py

```diff
@@ -0,0 +1,99 @@
+"""Constant folding of IR values.
```

Comment:
Should this file be in the `transform` folder?

Reply:
I put the file here since `transform` contains IR-to-IR tranforms, and this is basically an AST analysis pass, which seems closer to what is happening in `irbuild` overall.

Reply:
sounds good, then I don't have anything else blocking this from merging.